proposals/csharp-7.2/readonly-ref.md

```diff
@@ -40,105 +58,160 @@ static Vector3 Add (ref readonly Vector3 v1, ref readonly Vector3 v2)
     foo(ref v1.X);
 
     // OK
-    return new Vector3(v1.X +v2.X, v1.Y + v2.Y, v1.Z + v2.Z);
+    return new Vector3(v1.X + v2.X, v1.Y + v2.Y, v1.Z + v2.Z);
 }
 ```
 
-## Syntax
+`in` cannot be combined with other parameter passing modifiers such as `out`, `ref`, `params`. 
 
-One of proposed syntaxes is to use existing `in` keyword as a shorter form of `ref readonly`. It could be that both syntaxes are allowed or that we pick just one of them.
+## Use of `in` in signatures.
 
-While syntax is TBD, I will use `in` as it is significantly shorter than `ref readonly`.
+- `in` parameters and returns are allowed anywhere where ordinary byval parameters are allowed. This includes indexers, operators (including conversions), delegates, lambdas, local functions.
+ 
+> ```C#
+>  (in int x) => x                                                     // lambda expression  
+>  TValue this[in TKey index];                                         // indexer 
+>  public static Vector3 operator +(in Vector3 x, in Vector3 y) => ... // operator 
+>  ```
 
-```C# 
-static Vector3 Add (in Vector3 v1, in Vector3 v2)
-{
-    // OK
-    return new Vector3(v1.X +v2.X, v1.Y + v2.Y, v1.Z + v2.Z);
-}
-```
+- `in` would not be allowed in combination with `out` or with anything that `out` does not combine with.
```

Comment:
This is already covered in line 65 above.

Reply:
same line above is redundant.

---
In reply to: [154465874](https://github.com/dotnet/csharplang/pull/1165#discussion_r154465874) [](ancestors = 154465874)